code/modules/hivemind/wires.dm

```diff
@@ -0,0 +1,290 @@
+		master_node.add_wireweed(child)
+	spawn(1)
+		child.dir = get_dir(loc, target_turf) //actually this means nothing for wires, but need for animation
+		flick("spread_anim", child)
+		child.forceMove(target_turf)
+		update_icon()
+
+
+/obj/effect/plant/hivemind/proc/try_to_annihilate()
+	if(hive_mind_ai && master_node)
+		for(var/obj/machinery/machine_on_my_tile in loc)
+			var/can_annihilate = TRUE
+
+			//whitelist check
+			if(is_type_in_list(machine_on_my_tile, hive_mind_ai.restricted_machineries))
+				can_annihilate = FALSE
+
+			//annihilation is slow process, so it's take some time
+			//there we use our failure chance. Then it lower, then faster hivemind learn how to properly annihilate it
+			if(can_annihilate && prob(hive_mind_ai.failure_chance))
+				can_annihilate = FALSE
+				shake_the(machine_on_my_tile)
+				return
+
+			 //only one machine per turf
+			if(can_annihilate && !locate(/obj/structure/hivemind_machine) in loc)
+				annihilate(machine_on_my_tile)
+			//other will be... merged
+			else if(can_annihilate)
+				qdel(machine_on_my_tile)
+
+		//modular computers handling
+		var/obj/item/modular_computer/console/mod_comp = locate() in loc
+		if(mod_comp)
+			annihilate(mod_comp)
+
+		//dead bodies handling
+		for(var/mob/living/dead_body in loc)
+			if(dead_body.stat == DEAD)
+				annihilate(dead_body)
+
+
+/obj/effect/plant/hivemind/update_neighbors()
+	..()
+	update_connections()
+	update_icon()
+	update_openspace()
+
+
+/obj/effect/plant/hivemind/spread()
+	if(hive_mind_ai && master_node)
+		..()
+
+
+/obj/effect/plant/hivemind/life()
+	if(hive_mind_ai && master_node)
+		try_to_annihilate()
+		chem_handler()
+	else
+		//slow vanishing after node death
+		health -= 10
+		alpha = 255 * health/max_health
+		check_health()
+
+
+/obj/effect/plant/hivemind/is_mature()
+	return TRUE
+
+
+/obj/effect/plant/hivemind/refresh_icon()
+	overlays.Cut()
+	var/image/I
+	for(var/i = 1 to 4)
+		I = image(src.icon, "wires[wires_connections[i]]", dir = 1<<(i-1))
+		overlays += I
+	for(var/d in cardinal)
+		var/turf/T = get_step(loc, d)
+		if((locate(/obj/structure/window) in T) || istype(T, /turf/simulated/wall))
+			var/image/wall_hug_overlay = image(icon = src.icon, icon_state = "wall_hug", dir = d)
+			if (T.x < x)
+				wall_hug_overlay.pixel_x -= 32
+			else if (T.x > x)
+				wall_hug_overlay.pixel_x += 32
+			if (T.y < y)
+				wall_hug_overlay.pixel_y -= 32
+			else if (T.y > y)
+				wall_hug_overlay.pixel_y += 32
+			wall_hug_overlay.layer = ABOVE_WINDOW_LAYER
+			overlays += wall_hug_overlay
+
+
+/obj/effect/plant/hivemind/proc/update_connections(propagate = 0)
+	var/list/dirs = list()
+	for(var/obj/effect/plant/hivemind/W in range(1, src) - src)
+		if(propagate)
+			W.update_connections()
+			W.update_icon()
+		dirs += get_dir(src, W)
+
+	wires_connections = dirs_to_corner_states(dirs)
+
+
+//special check
+//machinery is allowed, door interaction also here
+/obj/effect/plant/hivemind/is_can_pass_special(turf/target)
```

Comment:
what is this wierd proc name all about? canpass is a thing that already existed, why do you need this?